tests/unit_tests/simpll/DFCLlvmIrTest.cpp

```diff
@@ -65,6 +75,107 @@ std::unique_ptr<Module> DFCLlvmIrTest::stringToModule(const char *llvm,
     return mod;
 }
 
+/// Returns instruction with the given name from the function
+/// or aborts program if the instruction with the name does not exist.
+Instruction *DFCLlvmIrTest::getInstByName(Function &fun, const char *name) {
+    auto value = fun.getValueSymbolTable()->lookup(name);
+    if (value) {
+        return dyn_cast<Instruction>(value);
+    }
+    errs() << "Error: Instruction '" << std::string(name)
+           << "' does not exist in '" << fun.getName().str() << "' function!\n";
+    std::abort();
+}
+
+/// Methods for testing DifferentialFunctionComparator using the TestComparator
+/// class. The methods cannot be called before CREATE_FROM_LLVM macro!!!
+
+/// Compares GEP instructions specified by name using cmpGEPs method.
+int DFCLlvmIrTest::testCmpGEPs(const char *name) {
+    checkPrepared();
+    auto gepL = dyn_cast<GEPOperator>(getInstByName(*FL, name));
+    auto gepR = dyn_cast<GEPOperator>(getInstByName(*FR, name));
+    checkValue("GEP instruction", name, gepL, gepR);
+    return DiffComp->testCmpGEPs(gepL, gepR);
+}
+
+/// Compares instructions specified by name using cmpOperations method.
+int DFCLlvmIrTest::testCmpOperations(const char *name,
+                                     bool &needToCmpOperands,
+                                     bool keepSN) {
+    checkPrepared();
+    auto instL = getInstByName(*FL, name);
+    auto instR = getInstByName(*FR, name);
+    return DiffComp->testCmpOperations(instL, instR, needToCmpOperands, keepSN);
```

Comment:
Shouldn't `checkValue` be here?
```suggestion
    checkValue(...);
    return DiffComp->testCmpOperations(instL, instR, needToCmpOperands, keepSN);
```
IIUC, `getInstByName` can return `nullptr` if the value is not an instruction.

Reply:
Yes, it can probably return `nullptr` (I am not sure what other things than instructions are saved in the `ValueSymbolTable` ),
1. I could either add the `checkValue` method or
2. update [the condition](https://github.com/PLukas2018/diffkemp/blob/23d8ecf217e1a741fc988801039debe7f87818aa/tests/unit_tests/simpll/DFCLlvmIrTest.cpp#L82) in the `getInstByName` method to be like this:
   ```cpp
       if (value && isa<Instruction>(value)) { // <- contains check that the value is an instruction
        return dyn_cast<Instruction>(value);
    }
    // aborting -- instruction with the given name does not exist
   ```
I like more the second option, because it makes sure that the function returns instruction and also it solves a possible problems in other `testCmp...` methods if the `getInstByName` returned a `nullptr`.

Reply:
Agreed, with a small adjustment of the code :-)
```cpp
    if (auto inst = dyn_cast<Instruction>(value)) {
     return inst;
 }
 // aborting -- instruction with the given name does not exist
```

Reply:
```cpp
...
    auto value = fun.getValueSymbolTable()->lookup(name);
    if (value) {
        if (auto inst = dyn_cast<Instruction>(value)) {
            return inst;
        }
    }
...
```

I also added ` if (value) {` because in case the symbol was not found, the `dyn_cast<Instruction>(value)` was asserting in the LLVM IR library because of `nullptr`, I think that it is more user friendly to leave the assertion on us to inform more detaily about the problem.